src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs

```diff
@@ -30,14 +31,13 @@ public void ThrowExceptionForHR_NoErrorInfo_ReturnsValidException(int errorCode)
                 Assert.Null(ex.InnerException);
                 Assert.Null(ex.HelpLink);
                 Assert.NotEmpty(ex.Message);
-                if (PlatformDetection.IsNetCore)
-                {
-                    Assert.Equal("System.Private.CoreLib", ex.Source);
-                }
-                else
-                {
-                    Assert.Equal("mscorlib", ex.Source);
-                }
+
+                string sourceMaybe = PlatformDetection.IsNetCore
```

Comment:
Is this going to work correctly on .NET Native? Mono?

Reply:
Though I guess this was the logic we already had?

Reply:
That is a great question. I have no idea. I assumed the logic of choice was correct. How would/could I verify this?

Reply:
> Though I guess this was the logic we already had?

Yep. I just added the option of the result being the current assembly. I had hoped this was already passing on other runtimes.